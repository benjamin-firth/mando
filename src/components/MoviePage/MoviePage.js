import React from 'react';
import logo from '../../logo.svg';


const MoviePage = () => {
  
  return (
    <section className="main">
      Waddup - Here's the container for movies
      <article className="movie">
        <div>
          <h2>Title</h2>
          <p>Episode #</p>
          <p>Release Year</p>
          <button type="button">View Characters</button>
        </div>
        <img src={logo} className="movie-picture" alt="Movie Poster" />
      </article>
      <article className="movie">
        <div>
          <h2>Title</h2>
          <p>Episode #</p>
          <p>Release Year</p>
          <button type="button">View Characters</button>
        </div>
        <img src={logo} className="movie-picture" alt="Movie Poster" />
      </article>
    </section>
  )
}

export default MoviePage;