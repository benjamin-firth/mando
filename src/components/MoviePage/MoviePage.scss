.movie-container {
  height: 90vh;
  overflow: scroll;
}