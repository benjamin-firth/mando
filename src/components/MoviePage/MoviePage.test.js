import React from 'react';
import { shallow } from 'enzyme';
import MoviePage from './MoviePage.js';

describe('MoviePage', () => {
  let wrapper;
  let mockMovie1;
  let mockMovie2;
  let mockMovie3;

  beforeEach(() => {
    mockMovie1 = { episode_id: 1 };
    mockMovie2 = { episode_id: 2 };
    mockMovie3 = { episode_id: 3 };
    wrapper = shallow(<
      MoviePage
      filmData={[mockMovie3, mockMovie2, mockMovie1]}
    />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should order the filmData based on episode id', () => {
    let expected = [mockMovie1, mockMovie2, mockMovie3];

    let orderedWrapper = shallow(<
      MoviePage
      filmData={expected}
    />);

    expect(wrapper).toEqual(orderedWrapper);
  })
})
