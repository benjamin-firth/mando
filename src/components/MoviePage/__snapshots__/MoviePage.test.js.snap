// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePage should match snapshot 1`] = `
<section
  className="movie-container"
>
  <Movie
    episode={1}
    image="https://media.proprofs.com/images/QM/user_images/1826446/1496915404.jpg"
    key="1"
  />
  <Movie
    episode={2}
    image="https://www.kickassfacts.com/wp-content/uploads/2015/12/Attack-of-the-Clones.jpg"
    key="2"
  />
  <Movie
    episode={3}
    image="https://i.kym-cdn.com/entries/icons/original/000/022/048/Revenge_of_the_Sith.jpg"
    key="3"
  />
</section>
`;
